src/document/DocumentAnnotator.ts

```diff
@@ -0,0 +1,82 @@
+// @flow
+import BaseAnnotator, { Options } from '../common/BaseAnnotator';
+import BaseManager from '../common/BaseManager';
+import RegionManager from '../region/RegionManager';
+import { ANNOTATOR_EVENT, CLASS_ANNOTATIONS_LOADED } from '../constants';
+
+export default class DocumentAnnotator extends BaseAnnotator {
+    annotatedEl?: HTMLElement;
+
+    annotations: Record<string, []> = {};
+
+    managers: Map<string, BaseManager[]> = new Map();
+
+    constructor(options: Options) {
+        super(options);
+
+        this.hydrate(options);
+    }
+
+    getPages(): HTMLElement[] {
+        // TODO: Inject page/container elements from Preview SDK rather than DOM?
+        return this.annotatedEl ? Array.from(this.annotatedEl.querySelectorAll('.page')) : [];
+    }
+
+    hydrate({ file }: Options): void {
+        this.api
+            .fetchAnnotations(file.file_version.id)
+            .then(() => {
+                // TODO: Normalize response, set in store, and render
+            })
+            .catch(error => {
+                this.emit(ANNOTATOR_EVENT.error, error);
+            });
+    }
+
+    init(scale: number): void {
+        super.init(scale);
+
+        if (!this.rootEl) {
+            return;
+        }
+
+        this.annotatedEl = this.rootEl.querySelector('.bp-doc') as HTMLElement;
+        this.annotatedEl.classList.add(CLASS_ANNOTATIONS_LOADED);
+
+        this.render();
+    }
+
+    render(): void {
+        this.getPages()
+            .filter(({ dataset }) => dataset.loaded && dataset.pageNumber)
+            .forEach(pageEl => this.renderPage(pageEl));
+    }
+
+    renderPage(pageEl: HTMLElement): void {
+        const { pageNumber = '1' } = pageEl.dataset;
+        const annotations = this.annotations[pageNumber];
+        const managers = this.managers.get(pageNumber) || [];
+
+        // Destroy any managers that were attached to page elements that no longer exist
+        if (managers.some(manager => !manager.exists(pageEl))) {
```

Comment:
This is interesting, if `pageEl` doesn't exist, wouldn't it always not exist? Can it be a simpler `if (!pageEl) { ...clean up managers... }`?

Also, `manager.exists(pageEl)` seemed confusing to me. Would it be better served as a domUtil or named as `pageExists`?

Reply:
The idea is to allow each manager to control its own root nodes, so I'm deferring to them to see if their root nodes exist. That code isn't present yet, though, since this is just scaffolding. The code in the managers would look something like this:

```
exists(pageEl) {
    return pageEl.contains(this.root);
}